src/actions/SignupAction.js

```diff
@@ -0,0 +1,54 @@
+import axios from 'axios';
+import { SIGNUP_PENDING, SIGNUP_SUCCESS, SIGNUP_FAIL } from './types';
+
+const appUrl = 'https://caret-bn-backend-staging.herokuapp.com/api/v1';
+
+export const signupAction = (userData) => (dispatch) => {
+  dispatch({
+    type: SIGNUP_PENDING,
+    payload: true,
+  });
+  axios.post(`${appUrl}/users/register`, userData)
+    .then((response) => {
+      dispatch({
```

Comment:
Kindly create an `isLoading` state in the component where you call this action to cater for loading